src/main/java/ua/tc/marketplace/service/CategoryService.java

```diff
@@ -2,42 +2,102 @@
 
 import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class CategoryService {
-
+    @Autowired
     private final CategoryRepository categoryRepository;
+    @Autowired
+    private final CategoryMapper categoryMapper;
+    @Autowired
+    private final ClassificationAttributeMapper classificationAttributeMapper;
+
+
+
+    public Page<CategoryDto> findAll(Pageable pageable) {
+        try {
+            Page<Category> categories = categoryRepository.findAll(pageable);
+            return categories.map(categoryMapper::convertToDto);
+        } catch (RuntimeException e) {
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Internal server error", e);
+        }
+    }
 
-    public Page<Category> findAll(Pageable pageable) {
-        return categoryRepository.findAll(pageable);
+    public CategoryDto findById(Long id) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(() -> new CategoryNotFoundException("Category with ID " + id + " not found"));
+        return categoryMapper.convertToDto(category);
     }
 
-    public Optional<Category> findById(Long id) {
-        return categoryRepository.findById(id);
+    public CategoryDto createCategory(CategoryDto categoryDto) {
+        try {
+            Category category = categoryMapper.toEntity(categoryDto);
+            Category savedCategory = categoryRepository.save(category);
+            return categoryMapper.convertToDto(savedCategory);
+        } catch (MappingException e) {
+
+            System.err.println("Error during mapping: " + e.getMessage());
+            throw new CategoryCreationException("Error creating category: mapping error", e);
+        } catch (Exception e) {
+
+            System.err.println("Error creating category: " + e.getMessage());
+            throw new CategoryCreationException("Error creating category", e);
+        }
     }
 
     public Category save(Category category) {
         return categoryRepository.save(category);
     }
 
-    public Category update(Long id, Category category) {
-        if (categoryRepository.existsById(id)) {
-            category.setId(id);
-            return categoryRepository.save(category);
-        } else {
-            throw new RuntimeException("Category not found");
+    public CategoryDto update(Long id, CategoryDto  categoryDto) {
+        try {
+            Category category = categoryMapper.toEntity(categoryDto);
+
+            Category updatedCategory = categoryRepository.findById(id)
+                    .map(existingCategory -> {
+                        category.setId(id);
+                        return categoryRepository.save(category);
+                    })
+                    .orElseThrow(() -> new CategoryNotFoundException("Category with ID " + id + " not found"));
+
+            return categoryMapper.convertToDto(updatedCategory);
+        } catch (CategoryNotFoundException e) {
```

Comment:
I think this try-catch block is redundant. I suggest to remove it. Or explain why do you use it here?